test-network-function/networking/suite.go

```diff
@@ -220,63 +255,76 @@ func runNetworkingTests(netsUnderTest map[string]netTestContext, count int) map[
 			for _, pod := range env.PodsUnderTest {
 				// The first container is used to get the network namespace
 				aContainerInPod := pod.ContainerList[0]
 				if _, ok := env.ContainersToExcludeFromConnectivityTests[aContainerInPod.ContainerIdentifier]; ok {
-					tnf.ClaimFilePrintf("Skipping pod %s because it is excluded from connectivity tests (default interface)", pod.Name)
+					tnf.ClaimFilePrintf("Skipping pod %s because it is excluded from all connectivity tests", pod.Name)
 					continue
 				}
 				netKey := "default" //nolint:goconst // only used once
-				defaultIPAddress := []string{pod.DefaultNetworkIPAddress}
+				defaultIPAddress := pod.DefaultNetworkIPAddresses
 				gomega.Expect(env).To(gomega.Not(gomega.BeNil()))
 				gomega.Expect(env.NodesUnderTest[aContainerInPod.NodeName]).To(gomega.Not(gomega.BeNil()))
 				gomega.Expect(env.NodesUnderTest[aContainerInPod.NodeName].DebugContainer.GetOc()).To(gomega.Not(gomega.BeNil()))
 				nodeOc := env.NodesUnderTest[aContainerInPod.NodeName].DebugContainer.GetOc()
-				processContainerIpsPerNet(&aContainerInPod.ContainerIdentifier, netKey, defaultIPAddress, netsUnderTest, nodeOc)
+				processContainerIpsPerNet(&aContainerInPod.ContainerIdentifier, netKey, defaultIPAddress, netsUnderTest, nodeOc, ipVersion)
 			}
-			badNets := runNetworkingTests(netsUnderTest, count)
+			badNets := runNetworkingTests(netsUnderTest, count, ipVersion)
 
 			if n := len(badNets); n > 0 {
 				log.Warnf("Failed nets: %+v", badNets)
-				ginkgo.Fail(fmt.Sprintf("%d nets failed the default network ping test.", n))
+				ginkgo.Fail(fmt.Sprintf("%d nets failed the default network %s ping test.", n, ipVersion))
 			}
 		})
 	})
 }
-func testMultusNetworkConnectivity(env *config.TestEnvironment, count int) {
+func testMultusNetworkConnectivity(env *config.TestEnvironment, count int, ipVersion string) {
+	identifier := identifiers.TestICMPv4ConnectivityMultusIdentifier
+	if ipVersion == IPv6 {
+		identifier = identifiers.TestICMPv6ConnectivityMultusIdentifier
+	}
 	ginkgo.When("Testing Multus network connectivity", func() {
-		testID := identifiers.XformToGinkgoItIdentifier(identifiers.TestICMPv4ConnectivityMultusIdentifier)
+		testID := identifiers.XformToGinkgoItIdentifier(identifier)
 		ginkgo.It(testID, ginkgo.Label(testID), func() {
 			netsUnderTest := make(map[string]netTestContext)
 			for _, pod := range env.PodsUnderTest {
 				// The first container is used to get the network namespace
 				aContainerInPod := pod.ContainerList[0]
 				if _, ok := env.ContainersToExcludeFromConnectivityTests[aContainerInPod.ContainerIdentifier]; ok {
-					tnf.ClaimFilePrintf("Skipping pod %s because it is excluded from connectivity tests (multus interface)", pod.Name)
+					tnf.ClaimFilePrintf("Skipping pod %s because it is excluded from all connectivity tests", pod.Name)
+					continue
+				} else if _, ok := env.ContainersToExcludeFromMultusConnectivityTests[aContainerInPod.ContainerIdentifier]; ok {
```

Comment:
you don't need the `else`, just `if` is enough

Reply:
fixed